### DISCUSSION TITLE
Resampling is not supported right?

### DISCUSSION THREAD

**Discussion by davidatsurge** [pdeljanov/Symphonia#131] [Q&A] [answered]
First off, thanks for this great library!

Second question: I wanted to confirm that I would have to use something like [rubato](https://github.com/HEnquist/rubato) to make sure I get samples at, say, a 16kHz rate, regardless of the sample rate of the audio being read, right?

**Comment by pdeljanov** [comment, marked as answer]
Hi @davidatsurge,

Glad you enjoy the library!

You're right, you'll need to resample to 16kHz manually to get that rate across all files. Symphonia decodes to whatever the rate of the file is. Rubato is a good choice, but there are also bindings to `libsamplerate`, `libspeex`, and `libsoxr` if you don't mind using C libraries.

**Reply by davidatsurge** [reply]
Thanks for the quick reply!

I am targeting wasm, so I imagine that staying in rust land would be easier.

While we're on that topic, are you aware of anyone targeting wasm with this library? I haven't actually tried compiling to wasm yet ...

To be clear, I'm not asking for JS bindings of a wasm module (which I think was pointed out as being in the works in the README), I'm just asking if the library compiles with wasm as the target at all. I'm writing some other parts of my project in wasm too so I don't need a JS binding for now.

**Reply by pdeljanov** [reply]
I think the [Ruffle](https://github.com/ruffle-rs/ruffle) folks are using Symphonia for audio decoding and one of their targets is WASM. Symphonia is all safe Rust and has very few dependencies, so it should be pretty straightforward hopefully. :)
